### PR TITLE
fix(nvmx): fixing async qpair connection bugs

### PR DESCRIPTION
**Pull request by dsavitskiy**

### CODE REVIEW

**Comment by dsavitskiy** [comment]
bors try

File: io-engine/src/bdev/nvmx/channel.rs [hunk 1]
```diff
@@ -513,17 +97,48 @@ pub struct NvmeIoChannelInner<'a> {
...
+    }
+
+    #[inline(always)]
+    pub(crate) fn qpair_mut(&mut self) -> &mut Option<QPair> {
+        &mut self.qpair
+    }
+
+    fn remove_qpair(&mut self) -> Option<QPair> {
+        if let Some(q) = &self.qpair {
+            debug!("removing qpair {q:p}", q = q.as_ptr());
```
Comment (tiagolobocastro) [inline io-engine/src/bdev/nvmx/channel.rs]: should this be trace?

File: io-engine/src/bdev/nvmx/qpair.rs [hunk 2]
```diff
@@ -0,0 +1,551 @@
...
+    fn to_string(&self) -> String {
+        match *self {
+            QPairState::Disconnected => "Disconnected",
+            #[cfg(feature = "spdk-async-qpair-connect")]
+            QPairState::Connecting => "Connecting",
+            QPairState::Connected => "Connected",
+            QPairState::Dropped => "Dropped",
+        }
+        .parse()
+        .unwrap()
```
Comment (tiagolobocastro) [inline io-engine/src/bdev/nvmx/qpair.rs]: is to_string() enough ?

**Review by tiagolobocastro — APPROVED** [review]

File: io-engine/src/bdev/nvmx/qpair.rs (hunk 2 above)
Comment (dsavitskiy) [reply io-engine/src/bdev/nvmx/qpair.rs]: ah, indeed. I just copied this piece of code over from the old implementation.

File: io-engine/src/bdev/nvmx/channel.rs (hunk 1 above)
Comment (dsavitskiy) [reply io-engine/src/bdev/nvmx/channel.rs]: yes

**Comment by dsavitskiy** [comment]
bors try

**Comment by dsavitskiy** [comment]
bors try

**Review by mtzaurus — APPROVED** [review]

**Comment by dsavitskiy** [comment]
bors merge

**Pull request merged**